engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -4021,10 +4021,23 @@ public NicProfile addVmToNetwork(final VirtualMachine vm, final Network network,
         }
     }
 
+    /**
+     * duplicated in {@see UserVmManagerImpl} for a {@see UserVmVO}
+     */
+    private void checkIfNetExistsForVM(VirtualMachine virtualMachine, Network network) {
```

Comment:
Minor: is there a place we can make this code accessible to both classes?

Reply:
I would like to look for that indeed, but there is still a secondary (corner-case) problem that needs fixing. Hence this intermediate solution. I'm thinking of a service in `QueryManager`